src/test/run-make/crate-attributes-with-cfg_attr/lib.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![cfg_attr(foo, crate_type="lib")]
```

Comment:
I checked compiletest and it skips `--crate-type` if the auxiliary file specifies `// no-prefer-dynamic` (although I don't know why it doesn't just fall back to lib in that case). I could use the duplicate `crate_type` attributes like that but it results in a warning about the first one being unused.